### PR TITLE
Task/14278 Exposure Submission: Test Type Selection

### PR DESCRIPTION
**Pull request by flxschmidt**
## Description
Provides a new screen Test Type Selection ("Art des Tests").

### Usage

Instantiate `TestTypeSelectionViewController` with the regarding View Model.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-14278

## Screenshots
Screenshot 2022-11-09 at 16 06 24

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift [hunk 1]
```diff
@@ -890,6 +896,48 @@ class ExposureSubmissionCoordinator: NSObject, RequiresAppDependencies {
...
+				self?.showNextScreenFromTestTypeSelectionScreen()
+			}, onDismiss: { [weak self] in
+				self?.parentViewController?.dismiss(animated: true)
+			}
+		)
+		
+		let footerViewController = FooterViewController(
+			FooterViewModel(
+				primaryButtonName: "Weiter",
+				primaryIdentifier: "todo",
```
Comment (30mar) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift]: Please add an actual accessibility identifier.

**Review by 30mar — COMMENTED** [review]
check my comment please

File: src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings [hunk 2]
```diff
@@ -929,6 +929,32 @@ Bei ausgeschalteter Hintergrundaktualisierung müssen Sie die App täglich aufru
 
 "ExposureSubmissionWarnOthers_continueButton" = "Einverstanden";
 
+/* Exposure Submission Warn Others Test Type Selection */
+
+"ExposureSubmission_TestTypeSelection_title" = "Art des Tests";
+
+"ExposureSubmission_TestTypeSelection_body" = "Bitte wählen Sie die Art des Tests aus, auf dessen Grundlage Sie warnen.";
+
+"ExposureSubmission_TestTypeSelection_optionSRSSelfTest_title" = "Antigen-Schnelltest nicht in der App registriert";
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings]: Can we have SRS before TestTypeSelection and also for other options on the bottom?
```suggestion
"ExposureSubmission_SRSTestTypeSelection_optionSelfTest_title" = "Antigen-Schnelltest nicht in der App registriert";
```

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift [hunk 3]
```diff
@@ -215,6 +215,12 @@ class ExposureSubmissionCoordinator: NSObject, RequiresAppDependencies {
 		navigationController?.topViewController?.view.endEditing(true)
 		navigationController?.pushViewController(vc, animated: true)
 	}
+	
+	private func present(_ vc: UIViewController, withNavigation: Bool = true) {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift]: We don't use shortcuts 

```suggestion
	private func present(_ viewController: UIViewController, withNavigation: Bool = true) {
```

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift [hunk 4]
```diff
@@ -215,6 +215,12 @@ class ExposureSubmissionCoordinator: NSObject, RequiresAppDependencies {
...
+		let navVC = NavigationControllerWithLargeTitle(rootViewController: vc)
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift:220]: navigationControllerWithLargeTitle

Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift]:
```suggestion
		let navigationControllerWithLargeTitle = NavigationControllerWithLargeTitle(rootViewController: vc)
```

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewController.swift [hunk 5]
```diff
@@ -0,0 +1,130 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import UIKit
+import OpenCombine
+
+class TestTypeSelectionViewController: DynamicTableViewController {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewController.swift]:
```suggestion
class SRSTestTypeSelectionViewController: DynamicTableViewController {
```

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift [hunk 6]
```diff
@@ -0,0 +1,134 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+import OpenCombine
+
+class TestTypeSelectionViewModel {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift]:
```suggestion
class SRSTestTypeSelectionViewModel {
```

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/__tests__/SAP_Internal_Stats_LinkCard+Mock.swift [hunk 7]
```diff
@@ -2,10 +2,8 @@
 // 🦠 Corona-Warn-App
 //
 
-@testable import ENA
-
 extension SAP_Internal_Stats_LinkCard {
-	static let rkiPandemicRadarURL: String = "https://www.rki.de/DE/Content/InfAZ/N/Neuartiges_Coronavirus/Situationsberichte/COVID-19-Trends/COVID-19-Trends.html?__blob=publicationFile#/home"
+	static let rkiPandemicRadarURLMock: String = "https://www.rki.de/DE/Content/InfAZ/N/Neuartiges_Coronavirus/Situationsberichte/COVID-19-Trends/COVID-19-Trends.html?__blob=publicationFile#/home"
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/__tests__/SAP_Internal_Stats_LinkCard+Mock.swift:6]: by mistake?

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/__tests__/SAP_Internal_Stats_LinkCard+Mock.swift [hunk 8]
```diff
@@ -17,7 +15,7 @@ extension SAP_Internal_Stats_LinkCard {
 	static func mock(
 		cardID: Int32 = 0,
 		updatedAt: Int64 = 0,
-		url: String = Self.rkiPandemicRadarURL
+		url: String = Self.rkiPandemicRadarURLMock
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/__tests__/SAP_Internal_Stats_LinkCard+Mock.swift:18]: same here

File: src/xcode/ENA/ENA/Source/View Helpers/AccessibilityIdentifiers.swift [hunk 9]
```diff
@@ -572,6 +572,15 @@ enum AccessibilityIdentifiers {
 
 		}
 
+		enum TestTypeSelection {
+			static let body = "AppStrings.ExposureSubmission.TestTypeSelection.body"
+			static let optionSRSSelfTest = "AppStrings.ExposureSubmission.TestTypeSelection.submissionType.srsSelfTest"
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/View Helpers/AccessibilityIdentifiers.swift]: Please remove SRS from here and below
```suggestion
			static let optionSelfTest = "AppStrings.ExposureSubmission.TestTypeSelection.submissionType.srsSelfTest"
```

File: src/xcode/ENA/ENA/Source/View Helpers/AccessibilityIdentifiers.swift [hunk 10]
```diff
@@ -572,6 +572,15 @@ enum AccessibilityIdentifiers {
 
 		}
 
+		enum TestTypeSelection {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/View Helpers/AccessibilityIdentifiers.swift]:
```suggestion
		enum SRSTestTypeSelection {
```

File: src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift [hunk 11]
```diff
@@ -182,6 +182,21 @@ enum AppStrings {
 
 			}
 		}
+
+		enum TestTypeSelection {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift]:
```suggestion
		enum SRSTestTypeSelection {
```

File: src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift [hunk 12]
```diff
@@ -182,6 +182,21 @@ enum AppStrings {
...
+			static let title = NSLocalizedString("ExposureSubmission_TestTypeSelection_title", comment: "")
+			static let body = NSLocalizedString("ExposureSubmission_TestTypeSelection_body", comment: "")
+			static let optionSRSSelfTestTitle = NSLocalizedString("ExposureSubmission_TestTypeSelection_optionSRSSelfTest_title", comment: "")
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift]: Also for the other options below

```suggestion
			static let optionSelfTestTitle = NSLocalizedString("ExposureSubmission_TestTypeSelection_optionSRSSelfTest_title", comment: "")
```

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift [hunk 13]
```diff
@@ -890,6 +896,48 @@ class ExposureSubmissionCoordinator: NSObject, RequiresAppDependencies {
 		push(topBottomContainerViewController)
 	}
 
+	// MARK: - Test Type Selection
+	
+	// to.do 14717: Remove default value."
+	/// Shows the Type of Tests screen
+	/// - Parameter preSelectSelfTest: If the process was started via self-report with self-test on the Manage Your Tests View, the Self-test entry shall be pre-selected. Otherwise, no entry shall be selected.
+	private func showTestTypeSelectionScreen(preSelectSelfTest: Bool = false) {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift]:
```suggestion
	private func showSRSTestTypeSelectionScreen(preSelectSelfTest: Bool = false) {
```

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewController.swift [hunk 14]
```diff
@@ -0,0 +1,130 @@
...
+	}
+	
+	@available(*, unavailable)
+	required init?(coder: NSCoder) {
+		fatalError("init(coder:) has not been implemented")
+	}
+	
+	// MARK: - Overrides
+
+    override func viewDidLoad() {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewController.swift]: Indentation is off or github issue?

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift [hunk 15]
```diff
@@ -890,6 +896,48 @@ class ExposureSubmissionCoordinator: NSObject, RequiresAppDependencies {
...
+		let vc = TestTypeSelectionViewController(
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift]:
```suggestion
		let srsTestTypeSelectionViewController = TestTypeSelectionViewController(
```

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift (hunk 13 above)
Comment (naveeddotio) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift]:
```suggestion
	private func showTestTypeSelectionScreen(isSelfTestTypePreSelected: Bool = false) {
```

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift [hunk 16]
```diff
@@ -0,0 +1,134 @@
...
+			return nil
+		}
+	}
+}
+
+fileprivate extension SAP_Internal_SubmissionPayload.SubmissionType {
+	var optionTitle: String {
+		switch self {
+		case .srsSelfTest:
+			return AppStrings.ExposureSubmission.TestTypeSelection.optionSRSSelfTestTitle
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift]: Would prefer this pattern
```suggestion
			return AppStrings.ExposureSubmission.SRSTestTypeSelection.optionSelfTestTitle
```

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift [hunk 17]
```diff
@@ -0,0 +1,134 @@
...
+			return AppStrings.ExposureSubmission.TestTypeSelection.optionSRSOtherTitle
+		default:
+			return ""
+		}
+	}
+	
+	var optionAccessibilityIdentifier: String? {
+		switch self {
+		case .srsSelfTest:
+			return AccessibilityIdentifiers.ExposureSubmission.TestTypeSelection.optionSRSSelfTest
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift]: Same here

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/__tests__/TestTypeSelectionViewModelTests.swift [hunk 18]
```diff
@@ -0,0 +1,43 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import XCTest
+@testable import ENA
+
+final class TestTypeSelectionViewModelTests: XCTestCase {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/__tests__/TestTypeSelectionViewModelTests.swift]:
```suggestion
final class SRSTestTypeSelectionViewModelTests: XCTestCase {
```

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift [hunk 19]
```diff
@@ -890,6 +896,48 @@ class ExposureSubmissionCoordinator: NSObject, RequiresAppDependencies {
...
+		
+		let topBottomContainerViewController = TopBottomContainerViewController(
+			topController: vc,
+			bottomController: footerViewController
+		)
+
+		present(topBottomContainerViewController)
+	}
+	
+	private func showNextScreenFromTestTypeSelectionScreen() {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift]:
```suggestion
	private func showSRSFlowNextScreen() {
```
```suggestion
	private func showNextScreenFromTestTypeSelectionScreen() {
```

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinatorModel.swift [hunk 20]
```diff
@@ -73,6 +73,13 @@ class ExposureSubmissionCoordinatorModel {
 		}
 	}
 	
+	/// Handle the storing of the selected submission type (only SRS types!).
+	/// - Parameter _ type: The selected submission type (SRS)
+	func submissionTypeSRSSelected(_ type: SAP_Internal_SubmissionPayload.SubmissionType) {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinatorModel.swift]:
```suggestion
	func storeSelectedSRSSubmissionType(_ type: SAP_Internal_SubmissionPayload.SubmissionType) {
```

**Review by naveeddotio — CHANGES_REQUESTED** [review]
Looks good, have some comments.

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift [hunk 21]
```diff
@@ -0,0 +1,134 @@
...
+		case .srsRegisteredRat:
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift:120]:
```suggestion
		case .srsRegisteredRAT:
```

**Review by 30mar — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift [hunk 22]
```diff
@@ -0,0 +1,134 @@
...
+			return AccessibilityIdentifiers.ExposureSubmission.TestTypeSelection.optionSRSRegisteredRat
+		case .srsRegisteredPcr:
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift:122]: Can we have RAT and PCR in caps here and other places?
```suggestion
		case .srsRegisteredPCR:
```

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift [hunk 23]
```diff
@@ -0,0 +1,134 @@
...
+	}
+	
+	/// The `SAP_Internal_SubmissionPayload.SubmissionType` that the user has selected in the list.
+	/// Is `nil`, as long as the user hasn't made a selection.
+	@OpenCombine.Published var selectedSubmissionType: SAP_Internal_SubmissionPayload.SubmissionType?
+	
+	// MARK: - Private
+
+	/// The order of the list entries shown.
+	private let submissionTypeForChoosing: [SAP_Internal_SubmissionPayload.SubmissionType] = [
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift]:
```suggestion
	private let submissionTypes: [SAP_Internal_SubmissionPayload.SubmissionType] = [
```

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift [hunk 24]
```diff
@@ -0,0 +1,134 @@
...
+//
+
+import Foundation
+import OpenCombine
+
+class TestTypeSelectionViewModel {
+	
+	// MARK: - Init
+	
+	init(preSelectSelfTest: Bool) {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift]:
```suggestion
	init(isSelfTestTypePreSelected: Bool) {
```

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift [hunk 25]
```diff
@@ -0,0 +1,134 @@
...
+		isInitialSelectionSelfTestSubmissionType = preSelectSelfTest
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift]:
```suggestion
		isSelfTestTypePreSelected = isSelfTestTypePreSelected
```

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewController.swift [hunk 26]
```diff
@@ -0,0 +1,130 @@
...
+		showWarnProcessCancelAlert()
+	}
+}
+
+
+// MARK: - Cell reuse identifiers.
+
+extension TestTypeSelectionViewController {
+	enum CustomCellReuseIdentifiers: String, TableViewCellReuseIdentifiers {
+		case optionGroupCell
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewController.swift]: I don't really get the name optionGroup, what does that mean?

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift [hunk 27]
```diff
@@ -0,0 +1,134 @@
...
+		.srsSelfTest,
+		.srsRegisteredPcr,
+		.srsUnregisteredPcr,
+		.srsRapidPcr,
+		.srsOther
+	]
+	
+	private var optionGroupSelectionSubscription: AnyCancellable?
+	
+	private let isInitialSelectionSelfTestSubmissionType: Bool
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift]:
```suggestion
	private let isSelfTestTypePreSelected: Bool
```

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift [hunk 28]
```diff
@@ -0,0 +1,134 @@
...
+	private let submissionTypeForChoosing: [SAP_Internal_SubmissionPayload.SubmissionType] = [
+		.srsRegisteredRat,
+		.srsSelfTest,
+		.srsRegisteredPcr,
+		.srsUnregisteredPcr,
+		.srsRapidPcr,
+		.srsOther
+	]
+	
+	private var optionGroupSelectionSubscription: AnyCancellable?
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift]: do we have a better name?

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift (hunk 3 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift]: Please be sure, that I made a fault before comment.
The shortcut is equal to previous implementation to be consistent. 
But I agree with you, I don't like that shortcuts as well. :)

```swift
// Current
private func push(_ vc: UIViewController) {
	navigationController?.topViewController?.view.endEditing(true)
	navigationController?.pushViewController(vc, animated: true)
}

private func present(_ vc: UIViewController, withNavigation: Bool = true) {
	let navVC = NavigationControllerWithLargeTitle(rootViewController: vc)
	navigationController?.topViewController?.view.endEditing(true)
	navigationController?.present(navVC, animated: true)
}
``` 

```swift
// Now
private func push(_ viewController: UIViewController) {
	navigationController?.topViewController?.view.endEditing(true)
	navigationController?.pushViewController(viewController, animated: true)
}

private func present(_ viewController: UIViewController, withNavigation: Bool = true) {
	let navigationControllerWithLargeTitle = NavigationControllerWithLargeTitle(rootViewController: viewController)
	navigationController?.topViewController?.view.endEditing(true)
	navigationController?.present(navigationControllerWithLargeTitle, animated: true)
}
```

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/__tests__/SAP_Internal_Stats_LinkCard+Mock.swift (hunk 7 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/__tests__/SAP_Internal_Stats_LinkCard+Mock.swift:6]: The reason I changed it to mock is, that we only use that for mock purposes. But it's the global extension of `SAP_Internal_Stats_LinkCard`. Of course, no part of that Task, but drops a Swift Lint warning, I saw yesterday that comes with a usage from Puneet in the ENA Target.

File: src/xcode/ENA/ENA/Source/View Helpers/AccessibilityIdentifiers.swift (hunk 9 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/View Helpers/AccessibilityIdentifiers.swift]: I would like to keep that, cause the actual type names begins with srs:

```swift
.srsRegisteredRat,
.srsSelfTest,
.srsRegisteredPcr,
.srsUnregisteredPcr,
.srsRapidPcr,
.srsOther
```

File: src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift (hunk 12 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift]: as mentioned above.

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewController.swift (hunk 14 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewController.swift]: gh issue

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewController.swift (hunk 26 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewController.swift]: It's a common name usage all over the app for those cells. You'll find 19 search results for that :)

You can image optionGroup like the radio option from HTML.

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift (hunk 28 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift]: I used same name of similar option group usages in the code

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift (hunk 1 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift]: good catch :D 

name is now localized. identifier is now default.

File: src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings (hunk 2 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Resources/Localization/de.lproj/Localizable.strings]: yes, good suggestion, changed everywhere

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift (hunk 4 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift]: done

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewController.swift (hunk 5 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewController.swift]: done

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift (hunk 6 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift]: done

File: src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/__tests__/SAP_Internal_Stats_LinkCard+Mock.swift (hunk 8 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/__tests__/SAP_Internal_Stats_LinkCard+Mock.swift:18]: see above

File: src/xcode/ENA/ENA/Source/View Helpers/AccessibilityIdentifiers.swift (hunk 10 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/View Helpers/AccessibilityIdentifiers.swift]: yes good

File: src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift (hunk 11 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/View Helpers/AppStrings.swift]: done

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift (hunk 13 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift]: good

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift (hunk 15 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift]: good

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift (hunk 16 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift]: done

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/__tests__/TestTypeSelectionViewModelTests.swift (hunk 18 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/__tests__/TestTypeSelectionViewModelTests.swift]: same done

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift (hunk 19 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift]: mmh taste, I've changed it to your suggestion.

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinatorModel.swift (hunk 20 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinatorModel.swift]: good, like it

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift (hunk 23 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift]: like

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift (hunk 24 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift]: like

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift (hunk 25 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift]: consistent, okay :)

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift (hunk 27 above)
Comment (flxschmidt) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/TestTypeSelection/TestTypeSelectionViewModel.swift]: done

**Review by naveeddotio — APPROVED** [review]

**Pull request merged**